students/kegan/session03/mailroom.py

```diff
@@ -24,28 +24,37 @@ def main():
     options = {
         '1': {
             'prompt': 'Send a Thank You',
-            'module': print_thank_you},
+            'function': print_thank_you},
         '2': {
             'prompt': 'Create a Report',
-            'module': print_report},
+            'function': print_report},
         '3': {
             'prompt': 'Write Thank Yous',
-            'module': write_thank_yous},
+            'function': write_thank_yous},
         '4': {
             'prompt': 'Quit',
-            'module': exit_program}}
+            'function': exit_program}}
     while True:
         print('Please choose from the following options:')
         prompt = '\n'.join(['{} {}'.format(
-            option, options[option]['prompt']) for option in options])
-        answer = input(prompt + '\n>')
+            num, option['prompt']) for num, option in options.items()])
+        answer = safe_input(prompt + '\n>')
         if answer in options:
-            options[answer]['module']()
+            options[answer]['function']()
+
+
+def safe_input(prompt):
+    try:
+        answer = input(prompt)
+    except (KeyboardInterrupt, EOFError):
+        return '4'  # returns the option that quits the program
```

Comment:
sneaky! well done.

in this case, it doesn't do much, but it would let you inject something before quitting the program -- save the data, etc...